src/DynamoCore/Scheduler/UpdateRenderPackageAsyncTask.cs

```diff
@@ -149,54 +149,122 @@ protected override void HandleTaskExecutionCore()
                 foreach (var el in mirrorData.GetElements())
                 {
                     if (el.IsCollection || el.Data is IGraphicItem)
                     {
-                        string newTag = tag + ":" + count;
-                        GetRenderPackagesFromMirrorData(outputPortId, el, newTag, displayLabels);
+                        string newLabel = labelKey + ":" + count;
+                        GetRenderPackagesFromMirrorDataImp(outputPortId, el, package, newLabel);
                     }
-                    count = count + 1;
+                    count += 1;
                 }
             }
             else
             {
-                var graphicItem = mirrorData.Data as IGraphicItem;
-                if (graphicItem == null)
+                if (!(mirrorData.Data is IGraphicItem graphicItem))
                 {
                     return;
                 }
 
-                var package = factory.CreateRenderPackage();
-                var packageWithTransform = package as ITransformable;
-                package.Description = tag;
+                GetRenderPackageFromGraphicItem(outputPortId, graphicItem, package, labelKey);
+            }
+        }
+
+        private void GetRenderPackageFromGraphicItem(Guid outputPortId, IGraphicItem graphicItem, IRenderPackage package, string labelKey)
+        {
+            var packageWithTransform = package as ITransformable;
+
+            try
+            {
+                var previousPointVertexCount = package.PointVertexCount;
+                var previousLineVertexCount = package.LineVertexCount;
+                var previousMeshVertexCount = package.MeshVertexCount;
 
-                try
+                //Plane tessellation needs to be handled here vs in LibG currently
+                if (graphicItem is Plane plane)
+                {
+                    CreatePlaneTessellation(package, plane);
+                }
+                else
                 {
-                    graphicItem.Tessellate(package, factory.TessellationParameters);
-                    if (package.MeshVertexColors.Any())
+                    try
                     {
-                        package.RequiresPerVertexColoration = true;
+                        graphicItem.Tessellate(package, factory.TessellationParameters);
                     }
-
-                    //If the package has a transform that is not the identity matrix
-                    //then set requiresCustomTransform to true.
-                    if (packageWithTransform != null && packageWithTransform.Transform.SequenceEqual(
-                        new double[] { 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1 }) == false)
+                    catch (LegacyRenderPackageMethodException)
                     {
-                        (packageWithTransform).RequiresCustomTransform = true;
+                        //At this point we have detected an implementation of Tessellate which is using legacy color methods
```

Comment:
We do not use the API's that throw `LegacyRenderPackageMethodException` in LibG so no changes are required.  Only some of the Color methods fall in this category.  `void ApplyPointVertexColors(byte[] colors)` for example as it replaces all the vertex colors in the render packages.  This would overwrite any color data that had previously been added.  `void AddPointVertexColor(byte red, byte green, byte blue, byte alpha)` is ok as it just continues to append data.